core/include/drivers/ls_dspi.h

```diff
@@ -0,0 +1,141 @@
+#define DSPI_MCR_CTXF	    0x00000800 /* Clear TX FIFO [20] */
+#define DSPI_MCR_CRXF	    0x00000400 /* Clear RX FIFO [21] */
+#define DPSI_XSPI	    0x00000008 /* Extended SPI Mode [28] */
+#define DSPI_MCR_PES	    0x00000002 /* Parity Error Stop [30] */
+#define DSPI_MCR_HALT	    0x00000001 /* Halt [31] */
+#define DPSI_ENABLE	    0x0
+#define DSPI_DISABLE	    0x1
+
+/* Transfer count */
+#define DSPI_TCR_SPI_TCNT(x) (((x) & 0x0000FFFF) << 16)
+
+/* Clock and transfer attributes */
+#define DSPI_CTAR_BRD	     0x80000000		/* Double Baud Rate [0] */
+#define DSPI_CTAR_FMSZ(x)    (((x) & 0x0F) << 27) /* Frame Size [1-4] */
+#define DSPI_CTAR_CPOL	     0x04000000		/* Clock Polarity [5] */
+#define DSPI_CTAR_CPHA	     0x02000000		/* Clock Phase [6] */
+#define DSPI_CTAR_LSBFE	     0x01000000		/* LSB First [7] */
+#define DSPI_CTAR_PCS_SCK(x) (((x) & 0x03) << 22) /* PCSSCK [8-9] */
+#define DSPI_CTAR_PA_SCK(x)  (((x) & 0x03) << 20) /* PASC [10-11] */
+#define DSPI_CTAR_P_DT(x)    (((x) & 0x03) << 18) /* PDT [12-13] */
+#define DSPI_CTAR_BRP(x)     \
+	(((x) & 0x03) << 16) /* Baud Rate Prescaler [14-15] */
+#define DSPI_CTAR_CS_SCK(x)  (((x) & 0x0F) << 12) /* CSSCK [16-19] */
+#define DSPI_CTAR_A_SCK(x)   (((x) & 0x0F) << 8)	/* ASC [20-23] */
+#define DSPI_CTAR_A_DT(x)    (((x) & 0x0F) << 4)	/* DT [24-27] */
+#define DSPI_CTAR_BR(x)	     ((x) & 0x0F)	/* Baud Rate Scaler [28-31] */
+
+/* Status */
+#define DSPI_SR_TXRXS	 0x40000000		  /* TX and RX Status [1] */
+#define DSPI_SR_TXCTR(x) \
+	(((x) & 0x0000F000) >> 12) /* TX FIFO Counter [16-19] */
+#define DSPI_SR_RXCTR(x) \
+	(((x) & 0x000000F0) >> 4)  /* RX FIFO Counter [24-27] */
+
+#define DSPI_DATA_8BIT	SHIFT_U32(8, 0)
+#define DSPI_DATA_16BIT SHIFT_U32(0xF, 0)
+
+#define DSPI_TFR_CONT	 (0x80000000)
+#define DSPI_TFR_CTAS(x) (((x) & 0x07) << 12)
+#define DSPI_TFR_PCS(x)	 (((1 << (x)) & 0x0000003f) << 16)
+#define DSPI_IDLE_DATA	 0x0
+
+/* tx/rx data wait timeout value, unit: us */
+#define DSPI_TXRX_WAIT_TIMEOUT 1000000
+
+/* Transfer Fifo */
+#define DSPI_TFR_TXDATA(x) (((x) & 0x0000FFFF))
+
+/* Bit definitions and macros for DRFR */
+#define DSPI_RFR_RXDATA(x) (((x) & 0x0000FFFF))
+
+/* CTAR register pre-configure mask */
+#define DSPI_CTAR_SET_MODE_MASK \
+	(DSPI_CTAR_FMSZ(15) | DSPI_CTAR_PCS_SCK(3) | DSPI_CTAR_PA_SCK(3) | \
+	 DSPI_CTAR_P_DT(3) | DSPI_CTAR_CS_SCK(15) | DSPI_CTAR_A_SCK(15) | \
+	 DSPI_CTAR_A_DT(15))
+
+/* SPI mode flags */
+#define SPI_CPHA      BIT(0) /* clock phase */
+#define SPI_CPOL      BIT(1) /* clock polarity */
+#define SPI_CS_HIGH   BIT(2) /* CS active high */
+#define SPI_LSB_FIRST BIT(3) /* per-word bits-on-wire */
+#define SPI_CONT      BIT(4) /* Continuous CS mode */
+
+/* default SCK frequency, unit: HZ */
+#define PLATFORM_CLK	      650000000
+#define DSPI_DEFAULT_SCK_FREQ 10000000
+#define DSPI_CLK_DIV	      4 /* prescaler divisor */
+#define DSPI_CLK	      (PLATFORM_CLK / DSPI_CLK_DIV) /* DSPI clock */
+#define CS_SPEED_MAX_HZ	      1000000	/* Slave max speed */
```

Comment:
Should all these macros be defined in the header file? Why not have them in the `ls_spi.c`?

Reply:
Move required macros to ls_dspi.c and rest are in ls_dspi.h which would be used in test API.
Not removed macros which are not being used currently considering the fact of enhancement in future.